app/routes/user/endorsements.server.ts

```diff
@@ -0,0 +1,236 @@
+
+    const db = await tables()
+    await db.circular_permission_requests.update({
+      Key: {
+        requestorSub: requestorSub,
+        endorserSub: this.#sub,
+      },
+      UpdateExpression: 'set #status = :status',
+      ExpressionAttributeNames: {
+        '#status': 'status',
+      },
+      ExpressionAttributeValues: {
+        ':status': status,
+      },
+    })
+  }
+
+  async deleteEndorsementRequest(data: FormData) {
+    const endorserSub = getFormDataString(data, 'endorserSub')
+    const db = await tables()
+    await db.circular_permission_requests.delete({
+      requestorSub: this.#sub,
+      endorserSub,
+    })
+  }
+
+  // Gets the current user's submitted endorsement requests
+  async getUsersEndorsements(): Promise<EndorsementRequest[]> {
+    const db = await tables()
+    const result = await db.circular_permission_requests.query({
+      KeyConditionExpression: '#requestorSub = :sub',
+      ExpressionAttributeNames: {
+        '#requestorSub': 'requestorSub',
+        '#requestorEmail': 'requestorEmail',
+        '#endorserSub': 'endorserSub',
+        '#endorserEmail': 'endorserEmail',
+        '#status': 'status',
+        '#created': 'created',
+      },
+      ExpressionAttributeValues: {
+        ':sub': this.#sub,
+      },
+      ProjectionExpression:
+        '#requestorSub, #requestorEmail, #endorserSub, #endorserEmail, #status, #created',
+    })
+
+    return result.Items as EndorsementRequest[]
+  }
+
+  // Requests awaiting the current user's approval
+  async getPendingEndorsementRequests(): Promise<EndorsementRequest[]> {
+    const db = await tables()
+    const result = await db.circular_permission_requests.query({
+      IndexName: 'byEndorserSub',
+      KeyConditionExpression: '#endorserSub = :endorserSub',
+      FilterExpression: '#status = :status',
+      ExpressionAttributeNames: {
+        '#endorserEmail': 'endorserEmail',
+        '#endorserSub': 'endorserSub',
+        '#status': 'status',
+        '#created': 'created',
+        '#requestorEmail': 'requestorEmail',
+        '#requestorSub': 'requestorSub',
+      },
+      ExpressionAttributeValues: {
+        ':endorserSub': this.#sub,
+        ':status': 'pending',
+      },
+      ProjectionExpression:
+        '#requestorSub, #requestorEmail, #endorserSub, #endorserEmail, #status, #created',
+    })
+
+    return result.Items as EndorsementRequest[]
+  }
+  //#endregion
+
+  async getEndorserSubFromCognito(email: string): Promise<string> {
+    const users = await getSubmitterUsersFromCognito()
+    if (!users) return ''
+    const potentialEndorsers = users.find(
+      (user) =>
+        user.Attributes?.find((attr) => attr.Name == 'email')?.Value == email
+    )
+    if (!potentialEndorsers) return ''
+
+    const userSub = potentialEndorsers.Attributes?.find(
+      (attr) => attr.Name == 'sub'
+    )?.Value
+    if (!userSub) return ''
+    return userSub
+  }
+}
```

Comment:
There may, in general, be multiple matching subs for a given email address.

Reply:
True, I am going to remove this from this layer. If we want to add a dropdown that fills with options as the user enters emails, maybe we could show multiple results in the dropdown along with the idp? like the dropdown text could read:
dakota@example.com (Facebook)
dakota@example.com (Google)
dakota@example.com (Email)
What are your thoughts?